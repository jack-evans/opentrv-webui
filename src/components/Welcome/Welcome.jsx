import React, { Component } from 'react'
import { Button } from 'carbon-components-react'

class Welcome extends Component {
  render () {
    return (
      <div className='Welcome'>
        <img src={require('./opentrv.jpg')} alt='OpenTRV logo' />
        <h2>Welcome to OpenTRV</h2>
        <h3>New User? Click Register to create an account</h3>
        <h3>Existing User? Login to view your TRV system</h3>
        <div className='Welcome__buttons'>
          <Button
            kind='secondary'
            className='Welcome__register-button'
            onClick={() => window.location.assign('/register')}
          >
            Register
          </Button>
          <Button
            className='Welcome__login-button'
            onClick={() => window.location.assign('/login')}
          >
            Login
          </Button>
        </div>
      </div>
    )
  }
}

export default Welcome
