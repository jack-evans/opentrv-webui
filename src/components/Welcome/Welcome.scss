
.Welcome {
  background-color: $ui-01;
  display: flex;
  height: 100vh;
  justify-content: center;
}