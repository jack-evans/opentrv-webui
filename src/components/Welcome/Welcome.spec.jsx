import React from 'react'
import Welcome from './Welcome'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Welcome.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<Welcome />)).toMatchSnapshot()
  })

  describe('when the register button is pressed', () => {
    let wrapper

    beforeEach(() => {
      global.location.assign = jest.fn()
      wrapper = mount(<Welcome />)
      wrapper.find('.Welcome__register-button').at(1).simulate('click')
    })

    afterEach(() => {
      jest.resetAllMocks()
      wrapper.unmount()
    })

    it('calls global.location.assign', () => {
      expect(global.location.assign).toHaveBeenCalledTimes(1)
    })

    it('calls global.location.assign with "/register"', () => {
      expect(global.location.assign).toBeCalledWith('/register')
    })
  })

  describe('when the login button is pressed', () => {
    let wrapper

    beforeEach(() => {
      global.location.assign = jest.fn()
      wrapper = mount(<Welcome />)
      wrapper.find('.Welcome__login-button').at(1).simulate('click')
    })

    afterEach(() => {
      jest.resetAllMocks()
      wrapper.unmount()
    })

    it('calls global.location.assign', () => {
      expect(global.location.assign).toHaveBeenCalledTimes(1)
    })

    it('calls global.location.assign with "/login"', () => {
      expect(global.location.assign).toBeCalledWith('/login')
    })
  })
})
