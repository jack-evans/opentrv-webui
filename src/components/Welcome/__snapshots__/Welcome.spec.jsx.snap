// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="OpenTRV logo"
          src="opentrv.jpg"
        />,
        <h2>
          Welcome to OpenTRV
        </h2>,
        <h3>
          New User? Click Register to create an account
        </h3>,
        <h3>
          Existing User? Login to view your TRV system
        </h3>,
        <div
          className="Welcome__buttons"
        >
          <Button
            className="Welcome__register-button"
            disabled={false}
            iconDescription="Provide icon description if icon is used"
            kind="secondary"
            onClick={[Function]}
            small={false}
            tabIndex={0}
            type="button"
          >
            Register
          </Button>
          <Button
            className="Welcome__login-button"
            disabled={false}
            iconDescription="Provide icon description if icon is used"
            kind="primary"
            onClick={[Function]}
            small={false}
            tabIndex={0}
            type="button"
          >
            Login
          </Button>
        </div>,
      ],
      "className": "Welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "OpenTRV logo",
          "src": "opentrv.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to OpenTRV",
        },
        "ref": null,
        "rendered": "Welcome to OpenTRV",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New User? Click Register to create an account",
        },
        "ref": null,
        "rendered": "New User? Click Register to create an account",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Existing User? Login to view your TRV system",
        },
        "ref": null,
        "rendered": "Existing User? Login to view your TRV system",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="Welcome__register-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="secondary"
              onClick={[Function]}
              small={false}
              tabIndex={0}
              type="button"
            >
              Register
            </Button>,
            <Button
              className="Welcome__login-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="primary"
              onClick={[Function]}
              small={false}
              tabIndex={0}
              type="button"
            >
              Login
            </Button>,
          ],
          "className": "Welcome__buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Register",
              "className": "Welcome__register-button",
              "disabled": false,
              "iconDescription": "Provide icon description if icon is used",
              "kind": "secondary",
              "onClick": [Function],
              "small": false,
              "tabIndex": 0,
              "type": "button",
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Login",
              "className": "Welcome__login-button",
              "disabled": false,
              "iconDescription": "Provide icon description if icon is used",
              "kind": "primary",
              "onClick": [Function],
              "small": false,
              "tabIndex": 0,
              "type": "button",
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="OpenTRV logo"
            src="opentrv.jpg"
          />,
          <h2>
            Welcome to OpenTRV
          </h2>,
          <h3>
            New User? Click Register to create an account
          </h3>,
          <h3>
            Existing User? Login to view your TRV system
          </h3>,
          <div
            className="Welcome__buttons"
          >
            <Button
              className="Welcome__register-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="secondary"
              onClick={[Function]}
              small={false}
              tabIndex={0}
              type="button"
            >
              Register
            </Button>
            <Button
              className="Welcome__login-button"
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              kind="primary"
              onClick={[Function]}
              small={false}
              tabIndex={0}
              type="button"
            >
              Login
            </Button>
          </div>,
        ],
        "className": "Welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "OpenTRV logo",
            "src": "opentrv.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to OpenTRV",
          },
          "ref": null,
          "rendered": "Welcome to OpenTRV",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New User? Click Register to create an account",
          },
          "ref": null,
          "rendered": "New User? Click Register to create an account",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Existing User? Login to view your TRV system",
          },
          "ref": null,
          "rendered": "Existing User? Login to view your TRV system",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="Welcome__register-button"
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="secondary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
                Register
              </Button>,
              <Button
                className="Welcome__login-button"
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="primary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
                Login
              </Button>,
            ],
            "className": "Welcome__buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Register",
                "className": "Welcome__register-button",
                "disabled": false,
                "iconDescription": "Provide icon description if icon is used",
                "kind": "secondary",
                "onClick": [Function],
                "small": false,
                "tabIndex": 0,
                "type": "button",
              },
              "ref": null,
              "rendered": "Register",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Login",
                "className": "Welcome__login-button",
                "disabled": false,
                "iconDescription": "Provide icon description if icon is used",
                "kind": "primary",
                "onClick": [Function],
                "small": false,
                "tabIndex": 0,
                "type": "button",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
